### ISSUE TITLE
Experiment's default plot option is not correctly displayed in SCXA

### ISSUE THREAD

**Issue by ke4** [ebi-gene-expression-group/atlas-web-single-cell#374]
We discussed the use case that should be implement for this with @upendrakumbham and @@pmb59.
There are 2 separate condition that we have to fullfill:

- If in the `priority` column in `scxa_dimension_reduction` table there is a **non zero** value (most probably 1) for a specific plot type for a specific experiment then the web app should show that one as default.
- If every value in the `priority` column in `scxa_dimension_reduction` table is **0**, then the web app should calculate a middle value for `Plot options` and display that one. For example: if we have 9 `Plot options` then we should display the 5th one.

Steps to reproduce:
1. Go to `Browse experiments` tab in `SCXA`.
2. Select [Transcriptional programs of circuit assembly in the Drosophila visual system](https://wwwdev.ebi.ac.uk/gxa/sc/experiments/E-GEOD-156455/results)

Expected result:
The middle `Plot option` is selected, as all the `priority` column in `scxa_dimension_reduction` table is **0**.

Actual result:
The 1st `Plot option` is selected.

You should also test it with `E-ANND-3` in your local environment. Please load that experiment into your local environment , if you have not done it, yet. The instructions to follow: [Add an AnnData experiment to our local SCXA](https://github.com/ebi-gene-expression-group/atlas-web-single-cell/issues/340)

Please see below how to reproduce the bug with that experiment.

-----
For historical reason I preserved the original ticket description and you can find it below.

Original title: **`E-ANND-3` experiment by default is not displaying plots in SCXA**

Steps to reproduce:
1. Go to `Browse experiments` tab in `SCXA`.
2. Select [Tabula Sapiens - Smart-seq2 dataset](http://localhost:8080/gxa/sc/experiments/E-ANND-3/results) experiment

**Expected result:**

Experiment's data should show up with the plots diagram.

**Actual result:**

An error message appears on the screen.

![Image](https://github.com/ebi-gene-expression-group/atlas-web-single-cell/assets/2183410/23017a83-024b-4a0e-bb49-c4ef2f91a1e5)

**Comment by ke4** [comment]
Here is the log from Solr cloud when it tries to load `E-ANND-3`:

[anndata_not_appearing.log](https://github.com/ebi-gene-expression-group/atlas-web-single-cell/files/13503727/anndata_not_appearing.log)

I need to investigate this more closely.

**Comment by ke4** [comment]
It looks like that I had an issue previously with my local `SCXA` environment regarding the data in my local DB.
I repopulated it and I managed to display `E-ANND-3` although there is a problem with the default view.
I attached the default view here:



![Image](https://github.com/ebi-gene-expression-group/atlas-web-single-cell/assets/2183410/23017a83-024b-4a0e-bb49-c4ef2f91a1e5)

**Comment by ke4** [comment]
If you go and select the only `plot option`, then the cell plots are correctly appear:



![Image](https://github.com/ebi-gene-expression-group/atlas-web-single-cell/assets/2183410/e11def10-5b0c-422c-8936-01166952f41c)

**Comment by ke4** [comment]
After some investigation it looks like the backend code is not correctly calculating the default `Plot options` for a specific plot type for a specific experiment.

The DB query that used by this method is not correct: https://github.com/ebi-gene-expression-group/atlas-web-single-cell/blob/7a6d042bc72e801452fd981627cd5d2bfa945c88/app/src/main/java/uk/ac/ebi/atlas/experimentpage/cellplot/CellPlotDao.java#L169

It does not take account that more than one values could be the same (in this case: `0`).

**Comment by lingyun1010** [comment]
I fixed the software bugs but in the meanwhile I found some errors in PostgreSQL database on dev.  I took a screenshot on what we have in the `scxa-dimension_reduction` search results. The issue is we have different formats on some experiments with `t-SNE` and `tsne`, `umap` and `UMAP` and mixture of three as shown in the screenshot.

![Image](https://github.com/ebi-gene-expression-group/atlas-web-single-cell/assets/33519183/7965ae2c-4337-44a5-91b6-7ba50e887894)
